### ISSUE TITLE
Execute method stuck in deadlock when called twice in a raw

### ISSUE THREAD

**Issue by SeeD-Seifer** [FubarDevelopment/restsharp.portable#25]
I've a 100% reproduce when client.Execute stuck in a deadlock if the method called twice in a raw.

Code example (this is not a working code, just an example of reproduce):
```
var t1 = client.Execute (request1);
var t2 = client.Execute (request2);
await Task.WaitAll (t1, t2);
// this point never reached
```

The reason: RequestGuard lock in a synchronous ways.

Using AsyncLock solves the problem. Examples of AsyncLock implementation could be found:
http://www.hanselman.com/blog/ComparingTwoTechniquesInNETAsynchronousCoordinationPrimitives.aspx and here
http://blogs.msdn.com/b/pfxteam/archive/2012/02/12/10266988.aspx

**Comment by fubar-coder** [comment]
Nice catch. I'll provide an update tomorrow. Thank you for reporting this issue.

**Comment by fubar-coder** [comment]
I've just released v2.4.2. Please tell me if the bug is fixed.

**Comment by SeeD-Seifer** [comment]
Thanks. New version works good.
